nft_ingester/src/program_transformers/bubblegum/collection_verification.rs

```diff
@@ -1,75 +1,145 @@
-                update_compressed_asset(txn, id_bytes.clone(), Some(seq), asset_to_update).await?;
+        let mut updated_metadata = metadata.clone();
+        updated_metadata.collection = Some(Collection {
+            key: collection.clone(),
+            verified: verify,
+        });
 
-                if verify {
-                    if let Some(Payload::SetAndVerifyCollection { collection }) =
-                        parsing_result.payload
-                    {
-                        let grouping = asset_grouping::ActiveModel {
-                            asset_id: Set(id_bytes.clone()),
-                            group_key: Set("collection".to_string()),
-                            group_value: Set(collection.to_string()),
-                            seq: Set(seq as i64),
-                            slot_updated: Set(bundle.slot as i64),
-                            ..Default::default()
-                        };
-                        let mut query = asset_grouping::Entity::insert(grouping)
-                            .on_conflict(
-                                OnConflict::columns([
-                                    asset_grouping::Column::AssetId,
-                                    asset_grouping::Column::GroupKey,
-                                ])
-                                .update_columns([
-                                    asset_grouping::Column::GroupKey,
-                                    asset_grouping::Column::GroupValue,
-                                    asset_grouping::Column::Seq,
-                                    asset_grouping::Column::SlotUpdated,
-                                ])
-                                .to_owned(),
-                            )
-                            .build(DbBackend::Postgres);
-                        query.sql = format!(
+        let updated_data_hash = hash_metadata(&updated_metadata)
+            .map(|e| bs58::encode(e).into_string())
+            .unwrap_or("".to_string())
+            .trim()
+            .to_string();
+
+        let asset_to_update = asset::ActiveModel {
+            id: Unchanged(id_bytes.clone()),
+            leaf: Set(Some(le.leaf_hash.to_vec())),
+            seq: Set(seq as i64),
+            data_hash: Set(Some(updated_data_hash.clone())), // todo remove clone
+            ..Default::default()
+        };
+        update_compressed_asset(txn, id_bytes.clone(), Some(seq), asset_to_update).await?;
+
+        // REMOVE: TESTING
+        let e = asset::Entity::find_by_id(id_bytes.clone())
+            .one(txn)
+            .await?
+            .unwrap();
+        let v: u8 = 1;
+        let dh = updated_data_hash.clone();
+        info!("creator hash: {}", e.creator_hash.clone().unwrap());
+        let computed_asset_hash = keccak::hashv(&[
+            &[v],
+            id_bytes.clone().as_ref(),
+            e.owner.as_ref().unwrap(),
+            e.delegate.unwrap_or(Default::default()).as_ref(),
```

Comment:
The problem could be because for the txs being used in testing, `owner == delegate`.  And then in our off-chain data store we do NOT save a delegate (we set it to NULL and you would end up getting `Default::default()` here).  But on chain would require the actual delegate pubkey to be used.

So I think you might need to do something like `e.delegate.unwrap_or(e.owner.as_ref().unwrap())`

Reply:
Ah ok thank you, let me try that!